shard-operator-parent/shard-operator-v1/src/main/java/com/redhat/service/smartevents/shard/operator/v1/controllers/BridgeExecutorController.java

```diff
@@ -89,31 +92,34 @@ public class BridgeExecutorController implements Reconciler<BridgeExecutor>,
     OperatorMetricsService metricsService;
 
     @Override
-    public List<EventSource> prepareEventSources(EventSourceContext<BridgeExecutor> eventSourceContext) {
-
-        List<EventSource> eventSources = new ArrayList<>();
-        eventSources.add(EventSourceFactory.buildSecretsInformer(kubernetesClient, LabelsBuilder.V1_OPERATOR_NAME, BridgeExecutor.COMPONENT_NAME));
-        eventSources.add(EventSourceFactory.buildDeploymentsInformer(kubernetesClient, LabelsBuilder.V1_OPERATOR_NAME, BridgeExecutor.COMPONENT_NAME));
-        eventSources.add(EventSourceFactory.buildServicesInformer(kubernetesClient, LabelsBuilder.V1_OPERATOR_NAME, BridgeExecutor.COMPONENT_NAME));
-        eventSources.add(EventSourceFactory.buildServicesMonitorInformer(kubernetesClient, LabelsBuilder.V1_OPERATOR_NAME, BridgeExecutor.COMPONENT_NAME));
-
-        return eventSources;
+    public Map<String, EventSource> prepareEventSources(EventSourceContext<BridgeExecutor> eventSourceContext) {
+        return EventSourceInitializer.nameEventSources(
+                EventSourceFactory.buildInformerFromOwnerReference(eventSourceContext, LabelsBuilder.V1_OPERATOR_NAME, BridgeExecutor.COMPONENT_NAME, Secret.class),
+                EventSourceFactory.buildInformerFromOwnerReference(eventSourceContext, LabelsBuilder.V1_OPERATOR_NAME, BridgeExecutor.COMPONENT_NAME, Service.class),
+                EventSourceFactory.buildInformerFromOwnerReference(eventSourceContext, LabelsBuilder.V1_OPERATOR_NAME, BridgeExecutor.COMPONENT_NAME, Deployment.class),
+                EventSourceFactory.buildInformerFromOwnerReference(eventSourceContext, LabelsBuilder.V1_OPERATOR_NAME, BridgeExecutor.COMPONENT_NAME, ServiceMonitor.class));
     }
 
     @Override
-    public UpdateControl<BridgeExecutor> reconcile(BridgeExecutor bridgeExecutor, Context context) {
+    public UpdateControl<BridgeExecutor> reconcile(BridgeExecutor bridgeExecutor, Context<BridgeExecutor> context) {
         LOGGER.info("Create or update BridgeProcessor: '{}' in namespace '{}'",
                 bridgeExecutor.getMetadata().getName(),
                 bridgeExecutor.getMetadata().getNamespace());
 
         BridgeExecutorStatus status = bridgeExecutor.getStatus();
 
+        // Always mark the AUGMENTING condition as TRUE at the beginning of the reconcile loop.
+        // If we are in timeout or in another dead path, override it as FALSE.
+        // If everything is already deployed and ready, the reconcile loop exits with no update.
+        status.markConditionTrue(ConditionTypeConstants.AUGMENTING);
```

Comment:
I don't understand if this change is related with the bump or not. It looks to me unrelated and in general an improvement.
Am I missing something?

Reply:
Basically the new version of the operator has a bug with the `max-attempts` configuration and caused a test to fail. Actually the test was failing because our code was bugged as well: the strategy to infer the status of the resource was implemented as 
```java
    public ManagedResourceStatus inferManagedResourceStatus() {
        if (isReady()) {
            return ManagedResourceStatus.READY;
        }
        if (isConditionTypeFalse(ConditionTypeConstants.READY)
                && !isConditionTypeTrue(SECRET_AVAILABLE)
                && !isConditionTypeTrue(IMAGE_NAME_CORRECT)
                && !isConditionTypeTrue(DEPLOYMENT_AVAILABLE)
                && !isConditionTypeTrue(SERVICE_AVAILABLE)
                && !isConditionTypeTrue(SERVICE_MONITOR_AVAILABLE)) {
            return ManagedResourceStatus.FAILED;
        }
        return ManagedResourceStatus.PROVISIONING;
    }
```

In short, in some cases it's possible (and expected) that the resource is NOT ready but some conditions are ready (i.e. `TRUE`). In these cases we do return that the resource is `FAILED`, but actually it might be `PROVISIONING`. With that implementation the strategy to infer the status of the resource was implicit and not covering all the possibilities: this is why I restored our old `AUGMENTING` condition to keep track that the resource is actually in a `PROVISIONING` status

Reply:
Ok thanks for the clarification :)